### DISCUSSION TITLE
CAN bus

### DISCUSSION THREAD

**Discussion by IhorNehrutsa** [CAP1Sup/Intellistep#105] [General]
Dear @ CAP1Sup, dear colleagues.
I think that there are not so many reasons to make a text interface over the CAN bus (as it is now).
More true is use the CAN bus upper layer interface. (There are enough of them.)

I extremely need CANOpen support on Intellistep
https://reprap.org/wiki/CANopen

There are several realisation of CANopen 
CANopenNode
Canfestival 
etc.
But they all take a lot of work.

I can't make up my mind in any way.
Perhaps you see a simpler option.

Can you help me?

**Comment by IhorNehrutsa** [comment]
This is an example

EM-CAN Series Stepper Drives
http://leadshine.com/series.aspx?type=products&category=stepper-products&producttype=stepper-drives&subtype=network-stepper-drives&series=em-can

**Comment by IhorNehrutsa** [comment]
CANopenNode CANopenEditor
https://github.com/CANopenNode/CANopenEditor/tree/build

**Comment by CAP1Sup** [comment]
So you're proposing this instead of Klipper support? I'd still really like to see Klipper support, but I can understand what you mean. Sounds good. Open a PR, then we'll go from there?
